include/OpenSpaceToolkit/Astrodynamics/Trajectory/Segment.hpp

```diff
@@ -75,6 +81,17 @@ class Segment
         Real computeDeltaV(const Real& aSpecificImpulse) const;
         Mass computeDeltaMass() const;
 
+        Map<Shared<Dynamics>, MatrixXd> getDynamicsContributions(const Shared<const Frame>& aFrameSPtr) const;
+        MatrixXd getDynamicsContribution(
+            const Shared<Dynamics>& aDynamicsSPtr,
+            const Shared<const Frame>& aFrameSPtr,
+            const Array<Shared<const CoordinatesSubset>>& aCoordinatesSubsetSPtrArray =
```

Comment:
would be great if this was a variable argument rather than an array

```
    (
            const Shared<Dynamics>& aDynamicsSPtr,
            const Shared<const Frame>& aFrameSPtr,
            Shared<const CoordinatesSubset> coordinatesSubsetSPtrs, ...
    )
```

Reply:
> Shared<const CoordinatesSubset> coordinatesSubsetSPtrs

Would allow a single coordinatesSubsetPtr though, is that what you would want? We can always use an Array of a single Pointer, nah?